test/integration/cluster/cluster_test.go

```diff
@@ -1,3 +1,5 @@
+// +build e2e
```

Comment:
This was meant to exclude the e2e tests from `make test-go` target, happy to change it to something else

Reply:
I worry about labeling these as e2e, since we'll want more comprehensive e2e tests that leverage cabpk and capd.

If we do want to exclude them from `make test-go`, then we'll need another prow job to validate these tests in CI.

Reply:
Sorry for the confusion, `make test-go` is usually run locally and as part of the 
`pull-cluster-api-test` CI target. 

These integration tests are only run in CI during the `pull-cluster-api-integration` target. Because now all packages are tested using `./...`, we'd need a way to preserve the current behavior. Maybe the `e2e` name is misleading, I can change to something like `integration-ci` or similar.